File: LegacyExamples/Nearables/AccelerometerExample/AccelerometerExample/DetailViewController.m
//
//  DetailViewController.m
//  AccelerometerExample
//
//  Created by Grzegorz Krukiewicz-Gacek on 05.01.2015.
//  Copyright (c) 2015 Estimote Inc. All rights reserved.
//

#import "DetailViewController.h"

@interface DetailViewController () <ESTNearableManagerDelegate>

@property (nonatomic, strong) ESTNearableManager *nearableManager;

@end

@implementation DetailViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    self.nearableManager = [ESTNearableManager new];
    self.nearableManager.delegate = self;
    
    [self.nearableManager startRangingForIdentifier:self.nearable.identifier];
    
    self.typeLabel.text = [ESTNearableDefinitions nameForType:self.nearable.type];
}

#pragma mark - ESTNearableManager delegate

- (void)nearableManager:(ESTNearableManager *)manager didRangeNearable:(ESTNearable *)nearable
{
    NSString *orientationString = @"";
    if (!nearable.isMoving)
    {
        switch (nearable.orientation)
        {
            case ESTNearableOrientationHorizontal:
                orientationString = @"Sticker is on its back";
                break;
            case ESTNearableOrientationHorizontalUpsideDown:
                orientationString = @"Sticker is on its front";
                break;
            case ESTNearableOrientationVertical:
                orientationString = @"Sticker is on its legs";
                break;
            case ESTNearableOrientationVerticalUpsideDown:
                orientationString = @"Sticker is on its head";
                break;
            case ESTNearableOrientationLeftSide:
                orientationString = @"Sticker is on its left side";
                break;
            case ESTNearableOrientationRightSide:
                orientationString = @"Sticker is on its right side";
                break;
            case ESTNearableOrientationUnknown:
                orientationString = @"Sticker orientation unknown";
                break;
                
            default:
                break;
        }
    }
    else
    {
        orientationString = @"Sticker is moving";
    }
    
    self.orientationLabel.text = orientationString;
    
    self.accelerometerLabel.text = [NSString stringWithFormat:@"x axis: %ldmG \n y axis: %ldmG \n z axis: %ldmG \n",
                                    (long) nearable.xAcceleration,
                                    (long) nearable.yAcceleration,
                                    (long) nearable.zAcceleration];
}

@end

File: Examples/nearables/AccelerometerExample-Swift/AccelerometerExample-Swift/DetailViewController.swift
//
//  DetailViewController.swift
//  AccelerometerExample-Swift
//
//  Created by Grzegorz Krukiewicz-Gacek on 05.01.2015.
//  Copyright (c) 2015 Estimote Inc. All rights reserved.
//

import UIKit

class DetailViewController: UIViewController, ESTNearableManagerDelegate {
    
    @IBOutlet weak var typeLabel: UILabel!
    @IBOutlet weak var orientationLabel: UILabel!
    @IBOutlet weak var accelerometerLabel: UILabel!
    
    var nearable:ESTNearable!
    var nearableManager:ESTNearableManager!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        nearableManager = ESTNearableManager()
        nearableManager.delegate = self
        nearableManager .startRangingForIdentifier(nearable.identifier)
        
        typeLabel.text = ESTNearableDefinitions.nameForType(nearable.type)
    }
    
    //MARK: - ESTNearableManager delegate
    
    func nearableManager(manager: ESTNearableManager!, didRangeNearable nearable: ESTNearable!) {
//        temperatureLabel.text = NSString(format: "%.1f°C", nearable.temperature)
        
        var orientationString:String
        if nearable.isMoving == false
        {
            switch nearable.orientation
            {
            case ESTNearableOrientation.Horizontal:
                orientationString = "Sticker is on its back"
            case ESTNearableOrientation.HorizontalUpsideDown:
                orientationString = "Sticker is on its front"
            case ESTNearableOrientation.Vertical:
                orientationString = "Sticker is on its legs"
            case ESTNearableOrientation.VerticalUpsideDown:
                orientationString = "Sticker is on its head"
            case ESTNearableOrientation.LeftSide:
                orientationString = "Sticker is on its left side"
            case ESTNearableOrientation.RightSide:
                orientationString = "Sticker is on its right side"
            case ESTNearableOrientation.Unknown:
                orientationString = "Sticker orientation unknown"
            }
        }
        else
        {
            orientationString = "Sticker is moving"
        }
        
        orientationLabel.text = orientationString
        
        self.accelerometerLabel.text = NSString(format: "x axis: %dmG \n y axis: %dmG \n z axis: %dmG \n",
        nearable.xAcceleration,
        nearable.yAcceleration,
        nearable.zAcceleration)
    }
}

